core/src/components/badge/badge.common.scss

```diff
@@ -56,10 +65,24 @@
   position: absolute;
 }
 
-:host(:not(.in-tab-button):empty.badge-vertical-top) {
-  top: 0;
+// Badge in Button
+// --------------------------------------------------
+
+:host([vertical]:not(.in-tab-button).in-button) {
+  // stylelint-disable-next-line property-disallowed-list
+  right: globals.$ion-space-050;
 }
 
-:host(:not(.in-tab-button):empty.badge-vertical-bottom) {
-  bottom: 0;
+:host(:not(:empty).in-button) {
+  min-width: globals.$ion-scale-400;
+  height: globals.$ion-scale-400;
+
+  font-size: globals.$ion-font-size-300;
+
+  line-height: globals.$ion-font-line-height-500;
+
+  ::slotted(ion-icon) {
+    width: globals.$ion-scale-300;
+    height: globals.$ion-scale-300;
+  }
```

Comment:
Tokens are only meant for ionic them only. Move these to the ionic file and use vars for native. Review how tab button implements this.